### PR TITLE
Do not set changed=True if the object did not change

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
Refreshed implementation of https://github.com/ansible/awx/issues/5474

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - API

##### AWX VERSION
```
9.3.0
```

##### ADDITIONAL INFORMATION
This won't work with check mode, check mode will say that it will change. But that's to be expected. For some changes, you don't know if the change will happen until you destructive try to make the change, and after you have tried, you will find out after the fact.

### CODE REVIEW

**Comment by AlanCoding** [comment]
so the issue here is that for values with $encrypted$ in them, this always says that it was not changed. That's not great.

**Comment by AlanCoding** [comment]
New behavior, as observed from the tests:

```
TASK [Create a valid SSH credential (old school)] ***********************************************************************************************************************************
changed: [localhost]

TASK [assert] ***********************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Create a valid SSH credential (new school)] ***********************************************************************************************************************************
[WARNING]: The field inputs of credential 101 has encrypted data and may inaccurately report task is changed.
changed: [localhost]

TASK [assert] ***********************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Create a valid SSH credential (new school)] ***********************************************************************************************************************************
changed: [localhost]

TASK [assert] ***********************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Create a valid SSH credential from lookup source (old school)] ****************************************************************************************************************
changed: [localhost]

TASK [assert] ***********************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Create a valid SSH credential from lookup source (new school)] ****************************************************************************************************************
[WARNING]: The field inputs of credential 102 has encrypted data and may inaccurately report task is changed.
changed: [localhost]

TASK [assert] ***********************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}
```

**Comment by AlanCoding** [comment]
The latest commit assures that the warning does not happen for creating new credentials - only when modifying an existing credential. This is the only case where the API doesn't give enough information to determine status. Okay, so we conservatively return True, and inform the user. I think this message should be good enough to get the users on the same page, and then if someone wants an enhancement to be sheepish about issuing PATCH requests, then they will have a much better shot a coherently forming that thought, and we can make an educated decision about any new options like that.

**Comment by AlanCoding** [comment]
wait, this has yet another unanticipated problem - it only considers dict fields for the $encrypted$ string, but several top-level fields can exhibit this case too, like `password`.

**Comment by AlanCoding** [comment]
came up with another case not handled with this, because it goes through a different code path. But I made an attempt to reuse as much of this logic as possible.

File: awx_collection/plugins/module_utils/tower_api.py [hunk 1]
```diff
@@ -574,6 +575,53 @@ def create_if_needed(self, existing_item, new_item, endpoint, on_create=None, it
         else:
             self.exit_json(**self.json_output)
 
+    def _encrypted_changed_warning(self, field, old, warning=False):
+        if not warning:
+            return
+        self.warn(
+            'The field {0} of {1} {2} has encrypted data and may inaccurately report task is changed.'.format(
+                field, old.get('type', 'unknown'), old.get('id', 'unknown')
+            ))
```
Comment (beeankha) [inline awx_collection/plugins/module_utils/tower_api.py:584]: 👍

**Review by beeankha — APPROVED** [review]

**Comment by AlanCoding** [comment]
Okay, I _think_ I'm done with the final checks, and ready to merge this now.

**Pull request merged**